File: packages/pdf-gen-py/pdf_gen/utils/logo.py
from typing import Any, Dict, Optional, Union

from .image import ImageSource, draw_image

LogoInput = Union[str, bytes, bytearray, Dict[str, Any]]


def resolve_logo(logo: Optional[LogoInput], default_size: float) -> Optional[Dict[str, Any]]:
    """Normalizes a template's `logo` field (bare path/bytes, or {src, width, height})."""
    if not logo:
        return None
    if isinstance(logo, dict):
        return {
            'src': logo['src'],
            'width': logo.get('width', default_size),
            'height': logo.get('height', default_size),
        }
    return {'src': logo, 'width': default_size, 'height': default_size}


__all__ = ['LogoInput', 'resolve_logo', 'draw_image', 'ImageSource']

File: packages/pdf-gen-js/src/utils/logo.ts
import { embedImage } from './image';

export type LogoInput = string | Buffer | { src: string | Buffer; width?: number; height?: number };

export interface ResolvedLogo {
  src: string | Buffer;
  width: number;
  height: number;
}

/** Normalizes a template's `logo` field (bare path/Buffer, or `{src,width,height}`) to a fixed shape. */
export function resolveLogo(logo: LogoInput | undefined, defaultSize: number): ResolvedLogo | null {
  if (!logo) return null;
  if (typeof logo === 'string' || Buffer.isBuffer(logo)) {
    return { src: logo, width: defaultSize, height: defaultSize };
  }
  return { src: logo.src, width: logo.width ?? defaultSize, height: logo.height ?? defaultSize };
}

export { embedImage };
